File: deamhome/src/main/java/sosteam/deamhome/domain/item/entity/Wishlist.java
package sosteam.deamhome.domain.item.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;
import sosteam.deamhome.domain.account.entity.Account;
import sosteam.deamhome.global.entity.LogEntity;

import java.util.List;

@Document
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Wishlist extends LogEntity {

    @DBRef(lazy = true)
    @Setter
    private Account account;

    @DBRef(lazy = true)
    private List<Item> items;

    public List<Item> addItem(Item item) {
        items.add(item);
        return this.items;
    }
}

File: src/main/kotlin/sosteam/deamhome/domain/item/entity/Wishlist.kt
package sosteam.deamhome.domain.item.entity

import lombok.Builder
import lombok.Setter
import org.springframework.data.mongodb.core.mapping.Document
import org.springframework.data.mongodb.core.mapping.DocumentReference
import sosteam.deamhome.domain.account.entity.Account
import sosteam.deamhome.global.entity.LogEntity

@Document
@Builder
class Wishlist(
	account: Account,
	items: List<Item>
) : LogEntity() {
	@DocumentReference(lazy = true)
	@Setter
	val account: Account = account
	
	@DocumentReference(lazy = true)
	val items: ArrayList<Item> = items as ArrayList<Item>
	
	fun addItem(item: Item): List<Item> {
		items.add(item)
		return items
	}
}